Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/VehicleRegistrationCertificate.hs

```diff
@@ -264,10 +266,12 @@ compareRegistrationDates actualDate providedDate =
 
 linkRCStatus :: (Id Person.Person, Id DM.Merchant) -> RCStatusReq -> Flow APISuccess
 linkRCStatus (driverId, merchantId) req@RCStatusReq {..} = do
+  mdriverFee <- DV.findPendingFeesByDriverId (cast driverId)
+  whenJust mdriverFee $ \driverFee -> do
+    when (driverFee.status == DF.PAYMENT_OVERDUE && req.isActivate) $ throwError (RCActivationFailedPaymentDue driverId.getId)
```

Comment:
It is better to check the `subscribed` flag = false in driverInformation as it will be possible to take rides even in case of multiple overdues, so RC change should also be allowed.